src/pages/index.js

```diff
@@ -1,109 +1,212 @@
-  return card.getView();
-}
+  profileModalCloseButton,
+  profileEditButton,
+  profileModalBox,
+  aviForm,
+} from "../utils/utils.js";
 
-function handleCardClick(cardData) {
-  previewImagePopup.open(cardData);
-}
+const userInfoEl = new UserInfo({
+  nameSelector: ".profile__title",
+  jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
+});
 
-//USER INFO
-const userInfo = new UserInfo({
-  userNameSelector: ".profile__title",
-  userTitleSelector: ".profile__description",
+/*---------- API ----------*/
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "92f7ebfa-9e8e-4fb8-b0a3-1cdaa15ccbe0",
+    "Content-Type": "application/json",
+  },
 });
+let sectionEl;
+let currentUserId;
+api
+  .getAppInfo()
+  .then(([userData, cardData]) => {
+    currentUserId = userData._id;
+    userInfoEl.setUserInfo({ name: userData.name, job: userData.about });
+    userInfoEl.setAvatarInfo(userData.avatar);
+
+    sectionEl = new Section(
+      {
+        items: cardData,
 
-//PopUpWithImage
-const previewImagePopup = new popupWithImage("#image-modal", handleImageClick);
-previewImagePopup.setEventListeners();
+        renderer: (data) => {
+          const card = createCard(data);
+          sectionEl.addItem(card);
+        },
+      },
+      ".cards__list"
+    );
+    sectionEl.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
 
-function handleImageClick(cardData) {
-  previewImagePopup.open(cardData);
+function createCard(data) {
+  const card = new Card({
+    data,
+    cardSelector: "#card-template",
+    currentUserId,
+    handlePreviewImage: () => {
+      imagePopup.open(data);
+    },
+    handleLikeCard: (shouldRemoveLike) => {
+      if (shouldRemoveLike) {
+        api
+          .removeLike(data._id)
+          .then((data) => {
+            card.updateLikes(data.isLiked);
+          })
+          .catch((err) => {
+            console.log(err);
+          });
+      } else {
+        api
+          .addLike(data._id)
+          .then((data) => {
+            card.updateLikes(data.isLiked);
+          })
+          .catch((err) => {
+            console.log(err);
+          });
+      }
+    },
+    handleDeleteCard: (cardId) => {
+      deletePopup.open();
+      deletePopup.setSubmitAction(() => {
+        deletePopup.renderLoading(true);
+        api
+          .deleteCard(cardId)
+          .then(() => {
+            card.deleteClick();
+            deletePopup.close();
+          })
+          .catch((err) => {
+            console.log(err);
+          })
+          .finally(() => {
+            deletePopup.renderLoading(false);
+          });
+      });
+    },
+  });
+  return card.generateCard();
 }
 
-//PopUpWithForm
-const profileEditPopup = new PopupWithForm(
-  "#profile-edit-modal",
-  handleEditProfileSubmit
+const addFormValidator = new FormValidator(addModalForm, formValidationConfig);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
const formValidators = {}

// enable validation
const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store a validator by the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```